crates/astria-sequencer/src/fees/state_ext.rs

```diff
@@ -565,6 +569,19 @@ pub(crate) trait StateWriteExt: StateWrite {
 
 impl<T: StateWrite> StateWriteExt for T {}
 
+/// Creates `abci::Event` of kind `tx.fees` for sequencer fee reporting
+fn construct_tx_fee_event(fee: &Fee) -> Event {
+    Event::new(
+        "tx.fees",
+        [
+            ("actionName", fee.action_name.to_string()).index(),
+            ("asset", fee.asset.to_string()).index(),
+            ("feeAmount", fee.amount.to_string()).index(),
+            ("sourceActionIndex", fee.source_action_index.to_string()).index(),
```

Comment:
I actually think we should change this name to `"positionInTransaction"`. I was part of the original review and had to do a double take what `sourceActionIndex` means.

Reply:
Followup issue here: https://github.com/astriaorg/astria/issues/1735